plugins/modules/cloud/xenserver/xenserver_guest.py

```diff
@@ -1818,13 +1884,13 @@ def main():
             type='list',
             elements='dict',
             options=dict(
-                name=dict(type='str', aliases=['name_label']),
+                name=dict(type='str', aliases=['name_label']),  # @TODO: required ?
                 mac=dict(type='str'),
-                type=dict(type='str', choices=['none', 'dhcp', 'static']),
+                type=dict(type='str', choices=['none', 'dhcp', 'static']),  # @TODO: required ?
                 ip=dict(type='str'),
                 netmask=dict(type='str'),
                 gateway=dict(type='str'),
-                type6=dict(type='str', choices=['none', 'dhcp', 'static']),
+                type6=dict(type='str', choices=['none', 'dhcp', 'static']),  # @TODO: required ?
```

Comment:
None are required. `name` is not required for existing network interfaces. For example you can change a MAC address only for some interface. `type` is not required because in some cases value is assumed depending on other parameters that are specified.

Reply:
done